testbed/lib/infra-stack.ts

```diff
@@ -0,0 +1,194 @@
+      ]
+    });
+
+    // The EKS cluster, without worker nodes as we'll add them later
+    const cluster = new eks.Cluster(this, 'Cluster', {
+      vpc: vpc,
+      role: clusterRole,
+      version: eks.KubernetesVersion.V1_21,
+      defaultCapacity: 0
+    });
+
+    // Worker node IAM role
+    const workerRole = new iam.Role(this, 'WorkerRole', {
+      assumedBy: new iam.ServicePrincipal('ec2.amazonaws.com'),
+      managedPolicies: [
+        iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonEKSWorkerNodePolicy'),
+        iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonEC2ContainerRegistryReadOnly'),
+        iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonEKS_CNI_Policy'),
+        iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonEKSVPCResourceController'), // Allows us to use Security Groups for pods
+        iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonSSMManagedInstanceCore')
+      ]
+    });
+  
+    cluster.awsAuth.addRoleMapping(workerRole, {
+        username: 'system:node:{{EC2PrivateDNSName}}',
+        groups: ['system:bootstrappers', 'system:nodes']
+    })
+
+    const sg = new SecurityGroup(this, "NodeSecurityGroup", {
+      description: "Worker Node Security Group",
+      vpc: vpc,
+    });
+    cluster.clusterSecurityGroup.addIngressRule(sg, ec2.Port.allTraffic())
+    sg.addIngressRule(cluster.clusterSecurityGroup, ec2.Port.allTraffic())
+    sg.addIngressRule(cluster.clusterSecurityGroup, ec2.Port.allTraffic())
+    sg.addEgressRule(ec2.Peer.anyIpv4(), ec2.Port.allTraffic())
+    sg.addEgressRule(ec2.Peer.anyIpv6(), ec2.Port.allTraffic())
+    Tags.of(sg).add('kit.sh/stack', super.stackName)
+    Tags.of(vpc).add('kit.sh/stack', super.stackName)
+
+    // Select the private subnets created in our VPC and place our worker nodes there
+    const privateSubnets = vpc.selectSubnets({
+      subnetType: ec2.SubnetType.PRIVATE_WITH_NAT,
+    });
+
+    const ng = cluster.addNodegroupCapacity('WorkerNodeGroup', {
+      subnets: privateSubnets,
+      nodeRole: workerRole,
+      minSize: 3,
+      maxSize: 3,
+      instanceTypes: [
+        new ec2.InstanceType('m5.large'),
+        new ec2.InstanceType('m5a.large'),
+        new ec2.InstanceType('m6i.large'),
+        new ec2.InstanceType('m6a.large'),
+        new ec2.InstanceType('t3.large'),
+        new ec2.InstanceType('t3a.large'),
+        new ec2.InstanceType('c5.large'),
+        new ec2.InstanceType('c5a.large'),
+        new ec2.InstanceType('c6i.large'),
+      ],
+      tags: {
+        "kit.sh/stack": super.stackName,
+      },
+      taints: [
+        {
+          effect: eks.TaintEffect.NO_SCHEDULE,
+          key: 'CriticalAddonsOnly',
+          value: 'true',
+        },
+      ],
+    });
+
+    // Setup Tekton test permissions
+
+    const ns = cluster.addManifest('tekton-tests-ns', {
+      apiVersion: 'v1',
+      kind: 'Namespace',
+      metadata: {
+          name: testNS
```

Comment:
Do you foresee multiple test namespaces? If we expect all tests within a single cluster to live in a single namespace, it might be kind of nice to just standardize on a well known namespace that's consistent across projects e.g. `test-suites`